duktape/src/main/java/com/squareup/duktape/Duktape.java

```diff
@@ -80,9 +106,30 @@ public String evaluate(String script) {
     }
   }
 
+  private static void checkSignatureSupported(Method method) {
+    if (!isSupportedArgType(method.getReturnType())) {
+      throw new UnsupportedOperationException(
+          String.format("Return type %s on %s is not supported",
+              method.getReturnType().toString(), method.getName()));
+    }
+    for (Class<?> parameterType : method.getParameterTypes()) {
+      if (!isSupportedArgType(parameterType)) {
+        throw new UnsupportedOperationException(
+            String.format("Parameter type %s on %s is not supported",
+                parameterType.toString(), method.getName()));
+      }
+    }
+  }
+
+  private static boolean isSupportedArgType(Class<?> parameterType) {
+    // TODO: support some more types.
```

Comment:
Weird to have a method named `arg type` with `void`, which can’t be used as an arg. (only a return type).